service/src/main/java/uk/gov/hmcts/reform/fpl/service/search/SearchService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.fpl.service.search;
+
+import lombok.RequiredArgsConstructor;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Map.of;
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class SearchService {
+
+    private final CoreCaseDataService coreCaseDataService;
+
+    public List<CaseDetails> search(String property, LocalDate day) {
+        requireNonNull(property);
+        requireNonNull(day);
+        return coreCaseDataService.searchCases(CASE_TYPE, dateQuery(property, day));
+    }
+
+    private String dateQuery(String property, LocalDate day) {
```

Comment:
How this deal with bank holidays? My understanding is according to the story it needs to look two days in front for business days only. so i.e. if I look for hearing on Friday I should be notify on hearing from Tuesday.

Reply:
see UpcomingHearingsFinder, it does:
LocalDate hearingDate = calendarService.getWorkingDayFrom(baseDate, noticeDaysBeforeHearing);
CalendarService is doing days calculation excluding weekends and bank holidays.

the only problem is when scheduler itself run on business day, in such situation mail will be send twice, on bank haoliday and on next non weekend day. I'm happy to address this issue as well

Reply:
Skipping job run on non working days has been added